src/main/java/edu/ksu/canvas/attendance/controller/SummaryController.java

```diff
@@ -100,42 +109,49 @@ private ModelAndView studentSummary(String sectionId, String studentId, boolean
         List<AttendanceSummaryModel> summaryForSections = isSimpleAttendance ?
                 reportService.getSimpleAttendanceSummaryReport(validatedSectionId) :
                 reportService.getAviationAttendanceSummaryReport(validatedSectionId);
-        List<LtiLaunchData.InstitutionRole> institutionRoles = canvasService.getRoles();
 
         student.getAttendances().sort(Comparator.comparing(Attendance::getDateOfClass).reversed());
 
-        summaryForSections.stream()
-                .flatMap(summary -> summary.getEntries().stream())
-                .filter(entry -> entry.getStudentId() == validatedStudentId)
-                .findFirst()
-                .ifPresent(entry ->  addAssignmentSummaryToPage(page, entry, isSimpleAttendance, student, assignment));
-        institutionRoles.stream()
-                .filter(institutionRole -> institutionRole.equals(LtiLaunchData.InstitutionRole.Learner))
-                .findFirst()
-                .ifPresent(role -> page.addObject("isStudent", true));
+        List<AttendanceSummaryModel.Entry> entries = new ArrayList<>();
 
-        page.addObject("sectionId", sectionId);
-        page.addObject("student", student);
-        page.addObject("summaryForm", makeupForm);
-        return page;
-    }
+        if (student.getSisUserId() != null) {
+            entries = summaryForSections.stream().flatMap(model -> model.getEntries().stream()).filter(entry -> student.getSisUserId().equals(entry.getSisUserId())).collect(Collectors.toList());
+        }
 
-    private void addAssignmentSummaryToPage(ModelAndView page, AttendanceSummaryModel.Entry entry, boolean isSimpleAttendance, AttendanceStudent student, AttendanceAssignment assignment) {
-        page.addObject("attendanceSummaryEntry", isSimpleAttendance ?
-            new AttendanceSummaryModel.Entry(entry.getCourseId(), entry.getSectionId(), entry.getStudentId(), entry.getSisUserId(), entry.getStudentName(), student.getDeleted(), entry.getTotalClassesTardy(), entry.getTotalClassesMissed(), entry.getTotalClassesExcused(), entry.getTotalClassesPresent())
-            : new AttendanceSummaryModel.Entry(entry.getCourseId(), entry.getSectionId(), entry.getStudentId(), entry.getSisUserId(), entry.getStudentName(), student.getDeleted(), entry.getSumMinutesMadeup(), entry.getRemainingMinutesMadeup(), entry.getSumMinutesMissed(), entry.getPercentCourseMissed()));
+        int totalPresentDays = 0, totalTardyDays  = 0, totalAbsentDays = 0, totalExcusedDays = 0;
+
+        for (AttendanceSummaryModel.Entry entry : entries) {
```

Comment:
The total presentDays calculation should probably be moved out of the controller and be unit tested.